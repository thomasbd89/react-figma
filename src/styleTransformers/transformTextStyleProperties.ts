import { GeometryProps, TextNodeProps } from '../types';
import { colorToRGB } from '../helpers/color';

export type TextStyleProperties = {
    color?: string;
};

interface TextProperties extends GeometryProps, TextNodeProps {}

export const transformTextStyleProperties = (style?: TextStyleProperties): TextProperties => {
    if (!style) {
        return {};
    }

    return {
        ...((style && style.color && { fills: [{ type: 'SOLID', color: colorToRGB(style.color) }] }) || {})
    };
};
