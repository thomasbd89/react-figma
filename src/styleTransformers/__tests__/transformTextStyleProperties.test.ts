import { transformTextStyleProperties } from '../transformTextStyleProperties';

describe('transformTextStyleProperties', () => {
    it('textAlign auto', () => {
        const result = transformTextStyleProperties({ textAlign: 'auto' });
        expect(result).toMatchSnapshot();
    });

    it('textAlign center', () => {
        const result = transformTextStyleProperties({ textAlign: 'center' });
        expect(result).toMatchSnapshot();
    });

    it('textAlign justify', () => {
        const result = transformTextStyleProperties({ textAlign: 'justify' });
        expect(result).toMatchSnapshot();
    });

    it('lineHeight "50%"', () => {
        const result = transformTextStyleProperties({ lineHeight: '50%' });
        expect(result).toMatchSnapshot();
    });

    it('lineHeight 1.4', () => {
        const result = transformTextStyleProperties({ lineHeight: 1.4 });
        expect(result).toMatchSnapshot();
    });

    it('lineHeight auto', () => {
        const result = transformTextStyleProperties({ lineHeight: 'auto' });
        expect(result).toMatchSnapshot();
    });

    it('letterSpacing "50%"', () => {
        const result = transformTextStyleProperties({ letterSpacing: '50%' });
        expect(result).toMatchSnapshot();
    });

    it('letterSpacing 1.4', () => {
        const result = transformTextStyleProperties({ letterSpacing: 1.4 });
        expect(result).toMatchSnapshot();
    });
});
