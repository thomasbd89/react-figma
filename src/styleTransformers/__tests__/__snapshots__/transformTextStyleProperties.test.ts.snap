// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformTextStyleProperties lineHeight "50%" 1`] = `
Object {
  "lineHeight": Object {
    "unit": "PERCENT",
    "value": 50,
  },
}
`;

exports[`transformTextStyleProperties lineHeight 1.4 1`] = `
Object {
  "lineHeight": Object {
    "unit": "PIXELS",
    "value": 1.4,
  },
}
`;

exports[`transformTextStyleProperties lineHeight auto 1`] = `
Object {
  "lineHeight": Object {
    "unit": "AUTO",
  },
}
`;

exports[`transformTextStyleProperties textAlign auto 1`] = `Object {}`;

exports[`transformTextStyleProperties textAlign center 1`] = `
Object {
  "textAlignHorizontal": "CENTER",
}
`;

exports[`transformTextStyleProperties textAlign justify 1`] = `
Object {
  "textAlignHorizontal": "JUSTIFIED",
}
`;
